src/Core/View.js

```diff
@@ -670,35 +677,65 @@ class View extends THREE.EventDispatcher {
+     * a specified coordinates, in this view.
      *
-     * @param {MouseEvent|Object} mouseOrEvt - A MouseEvent, or a screen
-     * coordinates.
-     * @param {number} [radius=3] - The precision of the picking, in pixels.
-     * @param {Layer[]} [where] - The layers to look into. If not specified, all
-     * `GeometryLayer` and `ColorLayer` layers of this view will be looked in.
+     * @param {Object} mouseOrEvt - Mouse position in window coordinates (from
+     * the top left corner of the window) or `MouseEvent` or `TouchEvent`.
+     * @param {number} [radius=3] - The picking will happen in a circle centered
+     * on mouseOrEvt. This is the radius of this circle, in pixels.
+     * @param {...ColorLayer|GeometryLayer|string} [where] - The layers to look
+     * into. If not specified, all {@link ColorLayer} and {@link GeometryLayer}
+     * layers of this view will be looked in.
      *
      * @return {Object} - An object, with a property per layer. For example,
      * looking for features on layers `wfsBuilding` and `wfsRoads` will give an
      * object like `{ wfsBuilding: [...], wfsRoads: [] }`. Each property is made
      * of an array, that can be empty or filled with found features.
      *
      * @example
-     * view.pickFeaturesAt({ x, y });
-     * view.pickFeaturesAt({ x, y }, 1, ['wfsBuilding']);
-     * view.pickFeaturesAt({ x, y }, 3, ['wfsBuilding', myLayer]);
+     * view.pickFeaturesOnColorLayersAt({ x, y });
+     * view.pickFeaturesOnColorLayersAt({ x, y }, 1, 'wfsBuilding');
+     * view.pickFeaturesOnColorLayersAt({ x, y }, 3, 'wfsBuilding', myLayer);
      */
-    pickFeaturesAt(mouseOrEvt, radius = 3, where = []) {
-        const result = {};
-        let layers = where.length == 0 ? this.getLayers(l => (l.isGeometryLayer || l.isColorLayer)) : where;
-        layers = layers.map((l) => {
-            const id = l.isLayer ? l.id : l;
-            result[id] = [];
-            return id;
+    pickFeaturesAt(mouseOrEvt, radius = 3, ...where) {
+        if (Array.isArray(where[0])) {
+            console.warn('Deprecated: the ...where argument of View#pickFeaturesAt should not be an array anymore, but a list: use the spread operator if needed.');
+            where = where[0];
+        }
+
+        const layers = [];
+
+        where = where.length == 0 ? this.getLayers(l => l.isColorLayer) : where;
+        where.forEach((l) => {
+            if (typeof l === 'string') {
+                l = this.getLayerById(l);
+            }
+
+            if (l && l.isColorLayer) {
+                layers.push(l.id);
+            }
         });
 
         // Get the mouse coordinates to the correct system
         const mouse = (mouseOrEvt instanceof Event) ? this.eventToViewCoords(mouseOrEvt, _eventCoords) : mouseOrEvt;
+        const objects = this.pickObjectsAt(mouse, radius, ...where);
+
+        const result = {};
+
+        if (objects.length > 0) {
```

Comment:
We could be return the data structure to `pickObjectsAt` to avoid this re-structuration ?
And unify the returns

Reply:
Yes we could, but `pickObjectsAt` doesn't pick only `GeometryLayers`, but `Object3D` too, so we can't really have the same data structure, because we lack layer ids. If it's okay to use the uuid from the objects though, it's okay with me.

Reply:
or `withoutLayer`...?